sdk/compute/Azure.ResourceManager.Compute/tests/Scenario/GalleryImageVersionContainerTests.cs

```diff
@@ -0,0 +1,216 @@
+            var vnetName = Recording.GenerateAssetName("testVNet-");
+            var subnetName = Recording.GenerateAssetName("testSubnet-");
+            ResourceIdentifier vnetId = $"{_resourceGroup.Id}/providers/Microsoft.Network/virtualNetworks/{vnetName}";
+            var addressSpaces = new Dictionary<string, object>()
+            {
+                { "addressPrefixes", new List<string>() { "10.0.0.0/16" } }
+            };
+            var subnet = new Dictionary<string, object>()
+            {
+                { "name", subnetName },
+                { "properties", new Dictionary<string, object>()
+                {
+                    { "addressPrefix", "10.0.2.0/24" }
+                } }
+            };
+            var subnets = new List<object>() { subnet };
+            var input = new GenericResourceData(DefaultLocation)
+            {
+                //Location = DefaultLocation,
+                Properties = new Dictionary<string, object>()
+                {
+                    { "addressSpace", addressSpaces },
+                    { "subnets", subnets }
+                }
+            };
+            return await _genericResourceContainer.CreateOrUpdateAsync(vnetId, input);
+        }
+
+        private async Task<GenericResource> CreateNetworkInterface(ResourceIdentifier subnetId)
+        {
+            _genericResourceContainer = DefaultSubscription.GetGenericResources();
+            var nicName = Recording.GenerateAssetName("testNic-");
+            ResourceIdentifier nicId = $"{_resourceGroup.Id}/providers/Microsoft.Network/networkInterfaces/{nicName}";
+            var input = new GenericResourceData(DefaultLocation)
+            {
+                Properties = new Dictionary<string, object>()
+                {
+                    { "ipConfigurations", new List<object>()
+                        {
+                            new Dictionary<string, object>()
+                            {
+                                { "name", "internal" },
+                                { "properties", new Dictionary<string, object>()
+                                    {
+                                        { "subnet", new Dictionary<string, object>() { { "id", subnetId.ToString() } } }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+            return await _genericResourceContainer.CreateOrUpdateAsync(nicId, input);
+        }
+
+        protected ResourceIdentifier GetSubnetId(GenericResource vnet)
+        {
+            var properties = vnet.Data.Properties as IDictionary<string, object>;
+            var subnets = properties["subnets"] as IEnumerable<object>;
+            var subnet = subnets.First() as IDictionary<string, object>;
+            return subnet["id"] as string;
+        }
```

Comment:
Looks like we are copying something from `VirtualMachineTestBase`. How about we just inherit from that class and use these functions directly?